test/distributed/test_data_parallel.py

```diff
@@ -531,10 +531,10 @@ def _test_scatter(self, tensor):
         self.assertEqual(result[0].get_device(), 0)
         self.assertEqual(result[1], x[2:])
         self.assertEqual(result[1].get_device(), 1)
-        grad = result[0].data.clone().fill_(2)
+        grad = result[0].clone().fill_(2)
```

Comment:
I think this ones needs a `.detach()` no? You don't want grad to share the history of result?